### ISSUE TITLE
[BUG] Ineffective vector item assignment

### ISSUE THREAD

**Issue by cmpute** [cython/cython#4078]
**Describe the bug**
Item assignment in a extension type member of C++ vector type is ineffective

**To Reproduce**
Library file `temp.pyx`
```cython
# distutils: language = c++

from libcpp.vector cimport vector

cdef class Vector:
    cdef public vector[int] data
    def __init__(self):
        self.data = [1,2,3]
```

Test script
```python
>>> from temp import Vector
>>> a = Vector()
>>> a.data
[1, 2, 3]
>>> a.data[0] = 2
>>> a.data
[1, 2, 3]
>>> a.data = [2,2,3]
>>> a.data
[2, 2, 3]
>>> exit()
```

**Expected behavior**
I expect the first item of `a` to be modified after `a.data[0]=2`

**Environment (please complete the following information):**
 - OS: Linux
 - Python version: Any
 - Cython version: 0.29.21

**Comment by da-woods** [comment]
When you expose it to Python as a public property it'll be converted to a list. You'll be inserting into the list, but there's no way of propagating the changes back to the vector. In doubt that we can do much to improve that, but it might be worth adding a warning message for properties that involve conversion to a mutable type

**Comment by cmpute** [comment]
Thanks for the clarification! I think one way to handle this by creating a list-like wrapper class for this kind of access instead of converting it to list. But it seems to be a big change, I think document this may also help!

**Comment by scoder** [comment]
Agreed that a warning for such a public property would be nice. PR welcome.
